puppet/ha-puppet/screenshot.js

```diff
@@ -264,12 +303,47 @@ export class Browser {
             polling: 100,
           },
         );
+
+        // Wait for all images and Lovelace cards to load
+        await page.waitForFunction(
+          () => {
+            // Check if all images are loaded
+            const images = Array.from(document.querySelectorAll("img"));
+            const allImagesLoaded = images.every(img => img.complete);
+
+            // Check if Lovelace UI is loaded (if we're on a Lovelace page)
+            const haEl = document.querySelector("home-assistant");
+            const mainEl = haEl?.shadowRoot?.querySelector("home-assistant-main");
+            const panelResolver = mainEl?.shadowRoot?.querySelector("partial-panel-resolver");
+            const panel = panelResolver?.children[0];
+
+            // Check if it's a Lovelace panel
+            if (panel && panel.tagName === "HUI-ROOT") {
```

Comment:
the panel inside `partial-panel-resolver` is never HUI-ROOT.

<img width="352" alt="image" src="https://github.com/user-attachments/assets/3eb493e6-3de4-453e-8c41-d14e5c0e7020" />

Also, make this a guard clause and return if you're not going to do anything.